app/src/main/java/com/example/recipeapp/Models/Parse/Preferences.java

```diff
@@ -0,0 +1,52 @@
+package com.example.recipeapp.Models.Parse;
+
+import com.example.recipeapp.Models.Recipe;
+import com.parse.ParseClassName;
+import com.parse.ParseException;
+import com.parse.ParseObject;
+
+@ParseClassName("Preferences")
+public class Preferences extends ParseObject {
+    public static final String KEY_NUMBER_OF_VOTES = "numberOfVotes";
+    public static final String KEY_AVG_TIME = "avgTime";
+    public static final String KEY_STD_TIME = "stdTime";
+    public static final String KEY_AVG_PRICE = "avgPrice";
+    public static final String KEY_AVG_HEALTH = "avgHealth";
+
+    public static final String KEY_USER_TASTE = "userTaste";
+    public static final String KEY_USER_DIET = "diet";
+    public static final String KEY_USER_CUISINE = "cuisine";
+
+    public void updatePreferences(Recipe recipe) {
+        increment(KEY_NUMBER_OF_VOTES);
+        try {
+            fetchIfNeeded();
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+
+        updateStd((float)recipe.getTimeToCook());
+
+        updateAverage(KEY_AVG_TIME, (double)recipe.getTimeToCook());
+        updateAverage(KEY_AVG_PRICE, recipe.getPricePerServing());
+        updateAverage(KEY_AVG_HEALTH, recipe.getHealthScore());
+
+//        ((Taste)getParseObject(KEY_USER_TASTE)).updateTaste(recipe);
+//        ((Diet)getParseObject(KEY_USER_DIET)).updateDiet(recipe);
+//        ((Cuisine)getParseObject(KEY_USER_CUISINE)).updateRecipe(recipe);
+    }
+
+    private void updateStd(float newTime) {
+        int n = getInt(KEY_NUMBER_OF_VOTES);
+        double sigma = getDouble(KEY_STD_TIME);
+        double avg = getDouble(KEY_AVG_TIME);
+        double newAvg = ((n - 1) * avg + newTime) / n;
+        put(KEY_STD_TIME, ((n - 1)*sigma + (newTime - avg)*(newTime - newAvg))/n);
+    }
```

Comment:
what are we doing in here, can we add a comment to the function as to how this is being used.